### PR TITLE
:sparkles: add doc generation to selflow-cli

### PR DESCRIPTION
**Pull request by Anthony-Jhoiro**
Added a command to generate the markdown documentation of the Selflow-CLI.

The documentation is generated based on command definitions created with [Cobra](https://github.com/spf13/cobra).

Documentations are generated in the `docs/ecosystem/cli` directory to be added to the Docusaurus documentation.

Usage:
```bash
# In the repo root
go run ./apps/selflow-cli gen-docs
# OR
nx run selflow-cli:gen-docs
```

### CODE REVIEW

File: apps/selflow-cli/selflow_gen_docs.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+				os.Exit(1)
+			}
+		},
+	}
+}
+
+const fileHeadingFormat = "---\nslug: %s\ntitle: \"⌨ %s\"\n---\n\n# ⌨️ `%s`\n\n"
+
+func generateDocs() error {
+	return doc.GenMarkdownTreeCustom(rootCmd, "./docs/ecosystem/cli", func(filepath string) string {
```
Comment (sylvain-lavazais) [inline apps/selflow-cli/selflow_gen_docs.go]: this line work on windows based file system ?

File: docs/getting-started.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 title: Getting Started
 ---
 
-The easiest way to get started with Selflow is to use the [Selflow CLI](./ecosystem/cli) and run the [Selflow Daemon](./ecosystem/selflow-daemon). You can follow [this documentation](./ecosystem/cli#installation).
+The easiest way to get started with Selflow is to use the [Selflow CLI](./ecosystem/cli/index.mdx) and run the [Selflow Daemon](./ecosystem/selflow-daemon). You can follow [this documentation](./ecosystem/cli#installation).
```
Comment (sylvain-lavazais) [inline docs/getting-started.md:5]: 1. the link of `selflow-cli` also exist on line line `29` of this file.
2. there is a typo on line 29 (suggestion from LanguageTool plugin):
image

**Review by sylvain-lavazais — APPROVED** [review]
Looking good 👍

**Comment by Anthony-Jhoiro** [comment]
Thanks for your review @lavazais-sylvain ! I made the changes you suggested. 

For the format of the generated files, I didn't find a simple way to use the library without recoding the generation functions. I might do it in the future but for now, I just added the `nx format` on the uncommitted files in the nx gen-docs task after the generation. What do you think about this change?

File: apps/selflow-cli/selflow_gen_docs.go (hunk 1 above)
Comment (Anthony-Jhoiro) [reply apps/selflow-cli/selflow_gen_docs.go]: Thanks! I missed that :penguin: !

**Review by sylvain-lavazais — APPROVED** [review]
👍

**Comment by sylvain-lavazais** [comment]
> Thanks for your review @lavazais-sylvain ! I made the changes you suggested.
> 
> For the format of the generated files, I didn't find a simple way to use the library without recoding the generation functions. I might do it in the future but for now, I just added the `nx format` on the uncommitted files in the nx gen-docs task after the generation. What do you think about this change?

well, it's better now 😃

**Pull request merged**